crates/turborepo-lib/src/daemon/server.rs

```diff
@@ -566,6 +573,43 @@ impl<PD: PackageDiscovery + Send + Sync + 'static> proto::turbod_server::Turbod
                 }
             })
     }
+
+    type PackageChangesStream = ReceiverStream<Result<proto::PackageChangeEvent, tonic::Status>>;
+
+    async fn package_changes(
+        &self,
+        _request: tonic::Request<proto::PackageChangesRequest>,
+    ) -> Result<tonic::Response<Self::PackageChangesStream>, tonic::Status> {
+        let mut package_changes_rx = self.file_watching.package_changes_watcher.package_changes();
```

Comment:
Can you add a comment about ordering here? How does the client know when changes will started to be accounted for? If the stream is open, and then the client makes a change, are we guaranteed to see a change event? Or is it possible that filewatching is still starting up?

Reply:
The first event we send is a rediscover event so in theory it shouldn't matter. Even if the client changes something, the first thing it's going to do after connecting to the daemon is build all of the packages

Reply:
Maybe I'm misunderstanding, but it doesn't look like that's what it's doing. The watcher task sends a rediscover message first, but there's no guarantee that a client will get that message, right?

Imagine we start the daemon, send the rediscover message, then make some changes, which triggers more package change notifications. _Then_ a client connects and wants to watch package changes. Is there a mechanism for that client to get the rediscover message?

Reply:
Ah yeah. I'll send a rediscover as the first message when we start up the stream.